pkg/pillar/hypervisor/hypervisor.go

```diff
@@ -43,14 +43,15 @@ type hypervisorDesc struct {
 var knownHypervisors = map[string]hypervisorDesc{
 	XenHypervisorName:        {constructor: newXen, dom0handle: "/proc/xen", hvTypeFileContent: "xen"},
 	KVMHypervisorName:        {constructor: newKvm, dom0handle: "/dev/kvm", hvTypeFileContent: "kvm"},
+	KubevirtHypervisorName:   {constructor: newKubevirt, dom0handle: "/dev/kvm", hvTypeFileContent: "kubevirt"},
```

Comment:
It would be a lot easier to read and understand this if we replace the dom0handle string with an enabled func. For the existing one that func would Stat a file and for kubervirt it would be base.IsHVTypeKube()

The functions can even be inline as in
constructor: newXen, enabled: func() bool { _, err := os.Stat("/proc/xen") return err == nil}, hvTypeFileContent: "xen"}